share/spice/gravatar/gravatar.js

```diff
@@ -20,77 +16,64 @@
                 sourceUrl: api_result.entry[0].profileUrl,
                 sourceIconUrl: 'http://gravatar.com/favicon.ico'
             },
+
             normalize: function() {
                 // Get the name of the user.
                 function getName(entry) {
-                    if(!entry.name || !entry.displayName) {
+                    if (!entry.name || !entry.displayName) {
                         var splitted = api_result.entry[0].profileUrl.split("/");
                         return splitted[splitted.length - 1];
                     }
-
-                    if(entry.name.formatted) {
+                    if (entry.name.formatted) {
                         return entry.name.formatted;
-                    } else if(entry.name.givenName && entry.name.familyName) {
+                    } else if (entry.name.givenName && entry.name.familyName) {
                         return entry.name.givenName + " " + entry.name.familyName;
                     }
                     return entry.displayName;
                 }
+                // Get current location
+                function getCurrentLocation(entry) {
+                    if (entry.currentLocation) {
+                        return entry.currentLocation;
+                    } else {
+                        return "No Location Specified";
```

Comment:
Let's rather not display anything if the info isn't there. Better to `return null`